### ISSUE TITLE
Reconfiguration of the nucleus when touched by user can be distracting

### ISSUE THREAD

**Issue by phetsims** [phetsims/build-an-atom#21]
This enhancement was requested by Emily M.  Currently, when a user touches a nucleus that they've built, the nucleus reconfigures itself as though the particle was removed.  This can be distracting, particularly on a touch-based interface, since students tend to poke at everything.  It would be better if the user had to move the particle a little ways before the nucleus was reconfigured.

**Comment by Nancy-Salpepi** [comment]
During today's design meeting with NS, KP, JB, AV, AP we decided to do the following:

1. Whatever is grabbed should move to the top. All associated values should change at that point. 
2. If it is straightforward, delay reconfiguring the nucleus until after dragging.

**Comment by jbphet** [comment]
I'm starting to investigate this and I want to make some notes for testing and comparison:
- @Nancy-Salpepi said above "[w]hatever is grabbed should move to the front".  This seems to happen for protons but not for neutrons.  I tested this on the current version on the website, which is v1.6.34, and I made helium-4.  If I then grab and drag around a proton, it's above the other nucleons in the z-order.  If I grab and move around a neutron, it is below the others.
- If we want to delay reconfiguring the nucleus until after a nucleon is dragged a bit, I think we will need to delay everything else too, such as updating the mass number, the nuclear stability, and the charge.  So, in other words, we shouldn't remove the nucleon from the modeled nucleus until after some dragging has occurred.

**Comment by jbphet** [comment]
I've implemented the idea described in my previous comment where the lowest-index z-layer is used for particles being dragged or residing in the bucket.  The code now also moves particles that start being dragged to the front of the z order in whatever layer they reside.  This seems to address the layering issue.

As for the reconfiguration issue, I changed the drag handler to wait to remove the nucleon from the nucleus until some dragging has occurred.  I have _not_, as of yet, added any sort of threshold distance to this, so any amount of dragging will work.  I did this because it's simpler in the code and because it seemed to me to work reasonably well.  We should decide if that's acceptable.

@Nancy-Salpepi - I'm assigning this to you to review the current behavior on main and determine whether you think it is now what we're after.

**Comment by Nancy-Salpepi** [comment]
@jbphet just to confirm, adding a delay in reconfiguration will also add a delay to all the values changing?

**Comment by jbphet** [comment]
> @jbphet just to confirm, adding a delay in reconfiguration will also add a delay to all the values changing?

Yes - doing it any other way would involve much more substantial code changes.

**Comment by AgustinVallejo** [comment]
I believe this was caused by a recent addition of zLayers for particles and we missed a spot. Ready for review and then mark for cherry-picking @Nancy-Salpepi
